### PR TITLE
LPS-190240 - Show Release section before Beta Section so that it opens first when visiting Instance Settings>>Feature Flags

### PR DESCRIPTION
**Pull request by ethib137**
https://liferay.atlassian.net/browse/LPS-190240

Configuration screens are ordered based on the configuration screen key [here](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/util/ConfigurationEntryRetrieverImpl.java#L383). By appending an index to the start of the key we are able to leverage this ordering to add the "Release" flags first. 

We are only concerned with the order of feature flags in the feature flags configuration, so I've isolated the changes here. The only downside of this is that it means that if we add a new feature flag type we need to add an update here as well.

### CODE REVIEW

**Comment by ethib137** [comment]
ci:stop

**Comment by ethib137** [comment]
ci:test:sf

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java [hunk 1]
```diff
@@ -48,8 +50,11 @@ public String getCategoryKey() {
 
 	@Override
 	public String getKey() {
-		return FeatureFlagConstants.getKey(
+		String featureFlagKey = FeatureFlagConstants.getKey(
 			_featureFlagType.toString(), getScope());
+
+		return _featureFlagsConfigurationOrder.indexOf(_featureFlagType) +
+			featureFlagKey;
```
Comment (thiago-buarqque) [inline modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java:57]: Declaring the variable `featureFlagKey` it's not necessary, it's single-use. It should be changed to:

```
return _featureFlagsConfigurationOrder.indexOf(_featureFlagType) +
			   FeatureFlagConstants.getKey(
				   _featureFlagType.toString(), getScope());
```

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java [hunk 2]
```diff
@@ -97,6 +102,11 @@ public void render(
 		}
 	}
 
+	private static final List<FeatureFlagType> _featureFlagsConfigurationOrder =
+		Arrays.asList(
+			FeatureFlagType.RELEASE, FeatureFlagType.BETA,
+			FeatureFlagType.DEPRECATION, FeatureFlagType.DEV);
+
 	private final FeatureFlagManager _featureFlagManager;
```
Comment (thiago-buarqque) [inline modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java:110]: Do you mind removing the `_featureFlagManager` in a new "SF" commit? I just noticed it's not used anywhere in the class.

**Review by thiago-buarqque — CHANGES_REQUESTED** [review]
Hey @ethib137 , the fix looks good. Just added some comments regarding source formatting.

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java (hunk 1 above)
Comment (ethib137) [reply modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java:57]: Yeah, that's what I thought too, but this one the source formatter made me change to this.

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java (hunk 2 above)
Comment (ethib137) [reply modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java:110]: Nice catch... removing it now.

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java (hunk 1 above)
Comment (thiago-buarqque) [reply modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/configuration/admin/display/FeatureFlagConfigurationScreen.java:55]: Oh well, that's weird. Okay then

**Comment by ethib137** [comment]
Updated the approach slightly based on recommendations from @drewbrokke.

**Comment by ethib137** [comment]
ci:test:sf

**Comment by ethib137** [comment]
ci:test:relevant

**Comment by ethib137** [comment]
ci:test:relevant

**Comment by ethib137** [comment]
Hey @john-co or @eltonccdantas could one of you guys check if these failures are related to the changes? Thanks.

**Comment by eltonccdantas** [comment]
Hey @ethib137, the failures are unrelated to your pull request. 🎉 

[Comparison Run](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1695296251473&testrayRunIdA=2427188346&testrayRunIdB=2428716625)

**Comment by eltonccdantas** [comment]
ci:forward:force

**Pull request closed without merging**